src/widget/systemtrayicon.cpp

```diff
@@ -107,16 +111,19 @@ SystemTrayIcon::SystemTrayIcon()
         {
             delete reinterpret_cast<QImage*>(image);
         };
-        QImage image(":/img/icon.png");
-        if (image.format() != QImage::Format_RGBA8888_Premultiplied)
-        image = image.convertToFormat(QImage::Format_RGBA8888_Premultiplied);
-        GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image.bits(), GDK_COLORSPACE_RGB, image.hasAlphaChannel(),
-                                 8, image.width(), image.height(),
-                                 image.bytesPerLine(), callbackFreeImage, &image);
+        QImage* image = new QImage(":/img/icon.png");
+        if (image->format() != QImage::Format_RGBA8888_Premultiplied)
+            *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
+        GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image->bits(), GDK_COLORSPACE_RGB, image->hasAlphaChannel(),
```

Comment:
Plus, using ```QPixmap``` is impractical in this case, as it doesn't provide an interface to extract raw data, like ```QImage::bits()```. Now, notice how the ```else```  block uses a ```QSystemTrayIcon```, the Qt-native solution, which doesn't really work — see [this bug](https://bugreports.qt.io/browse/QTBUG-31762), never fully resolved, AFAIK. However, for Plasma 5,we could go with a native solution starting from Qt5.4 (see [here](http://blog.martin-graesslin.com/blog/2014/06/where-are-my-systray-icons/)). I'll look into it.